db/models/articles.model.js

```diff
@@ -23,17 +23,31 @@ exports.selectAllComments = () => {
     })
 }
 
-exports.selectAllArticles = () => {
-    return db.query (
-        `SELECT author, title, article_id, topic, created_at, votes, article_img_url 
-        FROM articles
-        ORDER BY created_at DESC;`)
+exports.selectAllArticles = (topic, sort_by, order) => {
+    
+    if(!sort_by) {sort_by = 'created_at'}
+    if(!order) {order = 'desc'}
+    
+    let query = `
+        SELECT author, title, article_id, topic, created_at, votes, article_img_url,
+        CAST ((SELECT COUNT(*) FROM comments 
+        WHERE articles.article_id = comments.article_id) AS INTEGER)
+        AS comment_count
+        FROM articles`
+
+    if (topic) {
+        query += ` WHERE topic = '${topic}'`
```

Comment:
With the query and order logic here, we may be exposing ourselves to SQL injection. Whenever we are using template literals in SQL we will need to consider this. It is important to handle this before the query is passed to the database to avoid this risk.